core/block_validator.go

```diff
@@ -131,6 +131,7 @@ func (v *BlockValidator) ValidateState(block *types.Block, statedb *state.StateD
 	if root := statedb.IntermediateRoot(v.config.IsEIP158(header.Number)); header.Root != root {
 		return fmt.Errorf("invalid merkle root (remote: %x local: %x) dberr: %w", header.Root, root, statedb.Error())
 	}
+	statedb.Database().SaveTransitionState(header.Root)
```

Comment:
I'd expect this kind of logic right after the conversion logic, not necessarily after block header validations, but I guess is fine.

Reply:
that's where it would be if it didn't need the `header.Root` as a parameter. We need to confirm that the root is correct before saving the transition status.

Reply:
```suggestion
	// Verify that the advertised root is correct before 
	// it can be usedas an identifier for the conversion
	// status.
	statedb.Database().SaveTransitionState(header.Root)
```